src/Text/Pandoc/Writers/ICML.hs

```diff
@@ -527,11 +537,27 @@ parStyle opts style lst =
       let cont = inTags True "ParagraphStyleRange" attrs' content
       state $ \st -> (cont, st{ blockStyles = Set.insert stlStr $ blockStyles st })
 
+-- | Create the destination name
+makeDestName :: Text -> Text
+makeDestName name = "#" <> Text.replace " " "_" name
+
+-- | Create the markup for the content (incl. named destinations)
+-- |  NOTE: since we have no easy way to get actual named dests, we just create them for any short content blocks
+makeContent :: Doc Text -> Doc Text
+makeContent cont | isEmpty cont = empty
+              | length (show cont) < 25 = vcat [
+                    selfClosingTag "HyperlinkTextDestination"
+                      [("Self", "HyperlinkTextDestination/"<>makeDestName (render Nothing cont)), ("Name","Destination"), ("DestinationUniqueKey","1")]
+                    , inTagsSimple "Content" $ flush cont
+                  ]
+              | otherwise = inTagsSimple "Content" $ flush cont
```

Comment:
hm... not sure I understand what's going on here... why do add the `HyperlinkTextDestination` only if the content is less than 25 long?

Reply:
The comment above the method says it all
>-- |  NOTE: since we have no easy way to get actual named dests, we just create them for any short content blocks

AFAICT, there isn't anything in the Pandoc AST for a "destination", "bookmark" or "anchor", which is what I need here.   I know I could do it with a custom span attribute that I would read in filter, but here I need it in the "core" writer.   Instead, as a bit of a "hack", I just make any short content fragment as a link destination.

If you have a suggestion about how to set these bookmarks in the source - that I can find in the writer - I'm all ears!!

Reply:
yes, for that, the pandoc AST follows the HTML/markdown approach where you just can have an `id` attribute on any element, then link to it with `#`. (and yes, for headers, the `id` are auto-generated..) e.g.

```
$ pandoc -t native
# foo

bar [baz]{#anchor}

^D

[Header 1 ("foo",[],[]) [Str "foo"]
,Para [Str "bar",Space,Span ("anchor",[],[]) [Str "baz"]]]
```

Reply:
Thanks - that was 1/3 of what I needed.  The second third was figuring out how that shows up inside the writer itself, and in looking at the HTML and DocX writers, I worked that out (first param to the element!).

Now I have to figure out how to pass it through to the actual content writer...

Reply:
Accomplished!